### PR TITLE
feat(docs): clarify styled in docs

### PR DESCRIPTION
**Pull request by chasestarr**
### Description

- removes usage of `block` and `styled` in doc examples
- adds `useStyletron` documentation.
- applies `useStyletron` information to `styled` and `block` doc pages

### CODE REVIEW

File: documentation-site/examples/block/basic.js [hunk 1]
```diff
@@ -1,21 +1,14 @@
...
-        <Block color={colorString} key={colorString}>
-          {colorString}
-        </Block>,
-      );
-    }
-  }
-
-  return elements;
-};
+export default function() {
```
[the remark this answers is not in the document]
Comment (chasestarr) [reply documentation-site/examples/block/basic.js]: 🤷‍♂ I don't feel strongly about it

File: documentation-site/examples/block/basic.js [hunk 2]
```diff
@@ -1,21 +1,14 @@
...
+  return (
+    <React.Fragment>
+      <Block>first element</Block>
+      <Block paddingTop="100px">
```
[the remark this answers is not in the document]
Comment (chasestarr) [reply documentation-site/examples/block/basic.js]: My thought is that we can 'hand-wave' away the theme-related features in block. If there's no documentation for it, consumers will be less likely to use it. I think relying on string values to access theme keys is strictly worse for developer experience now that we have type-checking and autocomplete.

File: documentation-site/examples/icon/basic.js [hunk 3]
```diff
@@ -1,20 +1,23 @@
...
-    <ArrowLeft size={36} />
-    <ArrowRight size={36} />
-    <ArrowLeft size={36} />
-    <ArrowRight size={36} />
-  </Block>
-);
+export default () => {
+  const [useCss, theme] = useStyletron();
+  return (
+    <div className={useCss({color: theme.colors.warning})}>
```
[the remark this answers is not in the document]
Comment (chasestarr) [reply documentation-site/examples/icon/basic.js]: I don't agree that fewer characters equals better ergonomics. There are a couple benefits here that did not exist before: less custom api to learn, theme autocomplete, and type checking.

[the remark this answers is not in the document]
Comment (sandgraham) [reply documentation-site/examples/icon/basic.js]: why wouldn't autocomplete/type checking work with `Block` props if they are just static string value props?

re ergonomics: it is a fair bit less typing but also helps to reinforce the idea of building with the system values. ergonomics is about efficiency but also about helping folks think in the right model- which i think a theme primitive like Block/Box/etc does well.

it seems like some people love it and some people dislike the non standard api

File: documentation-site/examples/block/basic.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sandgraham) [reply documentation-site/examples/block/basic.js]: i think the warning at the top is prob enough to steer folks away from the pattern. but not documenting the main use case seems odd to me.

File: documentation-site/examples/icon/basic.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (chasestarr) [reply documentation-site/examples/icon/basic.js]: > why wouldn't autocomplete/type checking work with Block props if they are just static string value props?

Not saying that type-checking on `Block` props does not work, but it's more appropriate to what devs may be unfamiliar with - theme values. For example, the `color` prop will accept any string value. we can't assist in the case of misspelling `wanring`. Maybe not the biggest deal, but this error can be caught quickly now. We don't get any autocomplete with string values.

Screen Shot 2019-08-06 at 11 51 40 AM

[the remark this answers is not in the document]
Comment (chasestarr) [reply documentation-site/examples/icon/basic.js]: what do you mean by 'think in the right model'?

File: documentation-site/examples/block/basic.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (chasestarr) [reply documentation-site/examples/block/basic.js]: I'm happy to add the examples back if we see people are having issues

**Pull request merged**